tests/backend-storage/src/test/ServerStorage.test.ts

```diff
@@ -288,6 +288,39 @@ describe(`${ServerStorage.name}: ${serverStorage.constructor.name}`, () => {
         undefined
       );
 
+      const queriedReferences = await serverStorage.listObjects(
+        testDirectory.baseDirectory
+      );
+
+      expect(queriedReferences.length).to.be.equal(2);
+      const queriedReference1 = queriedReferences.find(
+        (ref) => ref.objectName === reference1.objectName
+      );
+      expect(queriedReference1).to.be.deep.equal(reference1);
+      const queriedReference2 = queriedReferences.find(
+        (ref) => ref.objectName === reference2.objectName
+      );
+      expect(queriedReference2).to.be.deep.equal(reference2);
+    });
+  });
+
+  // eslint-disable-next-line deprecation/deprecation
+  describe(`${serverStorage.list.name}()`, () => {
```

Comment:
`list` and `listObjects` should behave identically so it would be better to have the same test for both of them - the test logic could be extracted into a separate function that accepts a callback to test. The code won't be the prettiest but I think it is better than having duplicate code.